docs/about/contributing/modules.md

```diff
@@ -0,0 +1,406 @@
+
+**Key properties**:
+
+- `agent_id`: The Coder agent to execute the script on
+- `display_name`: Name shown in the Coder dashboard during execution
+- `run_on_start`: Execute when workspace starts (most common)
+- `run_on_stop`: Execute during workspace shutdown for cleanup
+- `script`: The actual shell script content to execute
+- `start_blocks_login`: Whether script completion is required before user access
+
+Example:
+
+```tf
+resource "coder_script" "install" {
+  agent_id          = var.agent_id
+  display_name      = "Install Tool"
+  script            = templatefile("${path.module}/run.sh", {
+    VERSION = var.version
+    ARCH    = data.coder_provisioner.me.arch
+  })
+  run_on_start      = true
+  start_blocks_login = true
+}
+```
+
+#### coder_app - User Interface
+
+[Create accessible applications in the Coder workspace interface](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/app), providing users with one-click access to tools and services.
+
+**When to use**:
+
+- Exposing web-based development tools (IDEs, dashboards)
+- Creating links to desktop applications via custom protocols
+- Providing access to running services
+
+**Key properties**:
+
+- `agent_id`: The Coder agent
+- `external`: `true` for protocol handlers, `false` for web apps
+- `group`: Name of a group that this app belongs to.
+- `healthcheck`: Monitor service availability
+- `order`: Control the display order of apps in the dashboard
+- `subdomain`: Access method for web apps
+- `url`: Service URL or protocol handler
+
+Example:
+
+```tf
+resource "coder_app" "service" {
+  agent_id = var.agent_id
+  slug     = "service"
+  url      = "http://localhost:${var.port}"
+
+  healthcheck {
+    url       = "http://localhost:${var.port}/api/status"  # Service-specific endpoint
+    interval  = 5
+    threshold = 6
+  }
+}
+```
+
+> **⚠️ Important**: Health check URLs are service-specific. Common paths include `/health`, `/healthz`, `/ping`, `/status`, or `/api/health`. Check your service's documentation or use the main service URL if no dedicated health endpoint exists.
+
+#### coder_env - Environment Variables
+
+[Set environment variables in workspace sessions](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/env) for tool configuration and authentication.
+
+**When to use**:
+
+- Configuring development tools and CLIs
+- Providing authentication tokens
+- Setting service endpoints
+
+Example:
+
+```tf
+resource "coder_env" "config" {
+  agent_id = var.agent_id
+  name     = "TOOL_CONFIG"
+  value    = jsonencode(var.config)
+}
+```
+
+#### Data Sources
+
+Access workspace context through Coder data sources:
+
+```tf
+data "coder_provisioner" "me" {}
+# Provides: arch, os for platform-specific logic
+
+data "coder_workspace" "me" {}
+# Provides: name, id, access_url, start_count
+
+data "coder_workspace_owner" "me" {}
+# Provides: name, email, full_name, groups of the workspace owner.
+```
+
+### Variable Design
```

Comment:
Why is it called design? Maybe we call it Module Inputs variable types. Also this is a terraform concept and not Coder so should also be able to link to some upstream docs.